### PR TITLE
Feature/OIDC integration2

### PR DESCRIPTION
**Pull request by sughics**
Grails4 upgrade and and OIDC in Integration

### CODE REVIEW

**Comment by sughics** [comment]
Required ansible configs have been added under feature branches. (To merge after this app is ready to deploy with OIDC)

 - https://github.com/AtlasOfLivingAustralia/ansible-inventories/tree/feature/oidc-bie-index
 - https://github.com/AtlasOfLivingAustralia/ala-install/tree/feature/oidc-bie-index

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#Tue Aug 08 13:13:22 AEST 2017
-grailsVersion=3.2.11
-grailsWrapperVersion=1.0.0
-gormVersion=6.0.12.RELEASE
-gradleWrapperVersion=3.4.1
+grailsVersion=4.0.13
```
Comment (yasima-csiro) [inline gradle.properties:1]: Please upgrade to 4.1.1 or to any secured version
More info: https://grails.org/blog/2022-07-18-rce-vulnerability.html

File: grails-app/controllers/bie/index/UrlMappings.groovy [hunk 2]
```diff
@@ -58,7 +58,7 @@ class UrlMappings {
 
         "/logout/logout" (controller: 'misc', action: 'logout')
 
-        "/"(view: "/index")
+        "/"(redirect: "/openapi")
```
Comment (yasima-csiro) [inline grails-app/controllers/bie/index/UrlMappings.groovy:61]: Why do we need this?

File: src/test/groovy/au/org/ala/bie/ImportServiceSpec.groovy [hunk 3]
```diff
@@ -1,68 +1,69 @@
...
+//                [ drUid: "dr130", boost: 10 ]
+//        ]
+//        importService.setConfiguration(grailsApplication.config)
+//
+//        def images = importService.loadImages(false)
+//        then:
+//        !images.isEmpty()
+//    }
+//
+//}
```
Comment (yasima-csiro) [inline src/test/groovy/au/org/ala/bie/ImportServiceSpec.groovy:69]: Shall we uncomment this and try to fix any issues occurring?

**Review by yasima-csiro — CHANGES_REQUESTED** [review]
Can you please check comments?

File: grails-app/controllers/bie/index/UrlMappings.groovy (hunk 2 above)
Comment (sughics) [reply grails-app/controllers/bie/index/UrlMappings.groovy:61]: This tires to maintain the behaviour of the current bie-index's index `/` [page](https://bie-ws.ala.org.au/ws/) which displays the swagger UI under the `/index` path. New changes replace the old swagger iplementation with openapi-1.0 for api spec definition and this change displayes the `/opeanapi` page for the `/`

Comment (yasima-csiro) [reply grails-app/controllers/bie/index/UrlMappings.groovy:61]: ah ok

File: src/test/groovy/au/org/ala/bie/ImportServiceSpec.groovy (hunk 3 above)
Comment (sughics) [reply src/test/groovy/au/org/ala/bie/ImportServiceSpec.groovy]: Updated

File: build.gradle [hunk 4]
```diff
@@ -102,6 +102,8 @@ dependencies {
     testCompile "org.seleniumhq.selenium:selenium-support:3.14.0"
     testRuntime "org.seleniumhq.selenium:selenium-chrome-driver:3.14.0"
     testRuntime "org.seleniumhq.selenium:selenium-firefox-driver:3.14.0"
+    // https://mavenlibs.com/maven/dependency/org.springframework/spring-messaging
```
Comment (yasima-csiro) [inline build.gradle:105]: Shall we remove this if it is necessary?

**Review by yasima-csiro — CHANGES_REQUESTED** [review]

**Review by yasima-csiro — APPROVED** [review]

**Pull request merged**